cli/provisionerrun.go

```diff
@@ -0,0 +1,78 @@
+package cli
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+	"golang.org/x/xerrors"
+
+	"cdr.dev/slog"
+	"cdr.dev/slog/sloggers/sloghuman"
+	"github.com/coder/coder/cli/cliflag"
+	"github.com/coder/coder/cli/cliui"
+)
+
+func provisionerRun() *cobra.Command {
+	var (
+		cacheDir           string
+		verbose            bool
+		useEchoProvisioner bool
+	)
+	cmd := &cobra.Command{
+		Use:   "run",
+		Short: "Run a standalone Coder provisioner",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+			notifyCtx, notifyStop := signal.NotifyContext(ctx, interruptSignals...)
+			defer notifyStop()
+
+			logger := slog.Make(sloghuman.Sink(cmd.ErrOrStderr()))
+			if verbose {
+				logger = logger.Leveled(slog.LevelDebug)
+			}
+
+			client, err := CreateClient(cmd)
+			if err != nil {
+				return err
+			}
+
+			errCh := make(chan error, 1)
+			provisionerDaemon, err := newProvisionerDaemon(ctx, client.ListenProvisionerDaemon, logger, cacheDir, errCh, useEchoProvisioner)
```

Comment:
I'm not saying it has to happen here, but since the provisioner has a name, consider using `filepath.Join(cacheDir, "provisionerd", name)`. Perhaps in `newProvisionerDaemon`.

This will allow multiple provisioners to run on the same machine without potentially breaking `terraform init`.